systemtest/src/main/java/io/strimzi/systemtest/resources/draincleaner/SetupDrainCleaner.java

```diff
@@ -34,6 +43,26 @@ public void applyInstallFiles(ExtensionContext extensionContext) {
             .filter(File::isFile)
             .collect(Collectors.toList());
 
+        // we need to create our own certificates before applying install-files
+        final SystemTestCertAndKey drainCleanerKeyPair = SystemTestCertManager
+            .generateRootCaCertAndKey("C=CZ, L=Prague, O=Strimzi Drain Cleaner, CN=StrimziDrainCleanerCA",
+                // add hostnames (i.e., SANs) to the certificate
+                new ASN1Encodable[] {
+                    new GeneralName(GeneralName.dNSName, "strimzi-drain-cleaner"),
+                    new GeneralName(GeneralName.dNSName, "strimzi-drain-cleaner.strimzi-drain-cleaner"),
+                    new GeneralName(GeneralName.dNSName, "strimzi-drain-cleaner.strimzi-drain-cleaner.svc"),
+                    new GeneralName(GeneralName.dNSName, "strimzi-drain-cleaner.strimzi-drain-cleaner.svc.cluster.local")
+                });
+        final CertAndKeyFiles drainCleanerKeyPairPemFormat = SystemTestCertManager.exportToPemFiles(drainCleanerKeyPair);
+
+        final Map<String, String> certsPaths = new HashMap<>();
+        certsPaths.put("tls.crt", drainCleanerKeyPairPemFormat.getCertPath());
+        certsPaths.put("tls.key", drainCleanerKeyPairPemFormat.getKeyPath());
+
+        final SecretBuilder customDrainCleanerSecretBuilder = SecretUtils.retrieveSecretBuilderFromFile(certsPaths,
+            "strimzi-drain-cleaner", "strimzi-drain-cleaner",
+            Collections.singletonMap("app", "strimzi-drain-cleaner"), "kubernetes.io/tls");
```

Comment:
You can go with `Constants.DRAIN_CLEANER_DEPLOYMENT_NAME` (or maybe rename it) instead of passing the hardcoded strings there.

Reply:
Done